<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

## Footnotes

## **Exercise 1 (Learning to interrupt)**

In [11]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

## **Exercise 2 (Resetting & Restarting the Kernal)**

In [2]:
print = 'PRINT'
print('Hello World')

TypeError: 'str' object is not callable

In [1]:
print('Hello World')

Hello World


## **Exercise 3 (Timing Stuff)**

In [1]:
%%timeit
squares = [i**2 for i in range(10)] 

1.34 μs ± 256 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [2]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

1.73 μs ± 155 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


`%%timeit` ran the codeblock 7 * 1,000,000 times and output the mean and standard deviation of each codeblock. The first method is on average faster (1.34 μs < 1.73 μs) than the second method, because it does not have the extra step of also turning the output into a list.